File: src/main/java/it/unisa/tirocinio/beans/VaultItemDTO.java
package it.unisa.tirocinio.beans;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VaultItemDTO {

    @JsonProperty("item")
    private VaultItem item;
    @JsonProperty("ownerId")
    private String ownerId;
}

File: vaultitem-service/src/main/kotlin/it/unisa/vaultservice/vaultitemservice/beans/VaultItemDTO.kt
package it.unisa.vaultservice.vaultitemservice.beans

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.*
import java.time.LocalDate


@Data
@NoArgsConstructor
class VaultItemDTO
    (
    @JsonProperty("id") var id: String?,
    @JsonProperty("idCardNumber") var idCardNumber: String,
    @JsonProperty("resourceURI") var resourceURI: String,
    @JsonProperty("itemName") var itemName: String,
    @JsonProperty("creationDate") var creationDate: LocalDate,
    @JsonProperty("ownerId") var ownerId: String
)